### PR TITLE
Reenable scale unification

### PR DESCRIPTION
**Pull request by vshampor**
### Changes
Restored the original, pre-#1778 logic of scale unification and added missing tests for the logic. Added `BatchNorm` as a quantizable operation (asymmetric, per-channel) to the CPU HW config to handle cases like densenet where batch norm is the first operation in a branch.

### Reason for changes
Scales are currently not correctly unified in cases such as #2195.

### Related tickets
N/A

### Tests
tests/common/quantization/test_quantizer_setup.py
tests/**/quantization/test_unified_scales.py

Fixes: #2195

### CODE REVIEW

**Comment by vshampor** [comment]
I put this out of the draft state to trigger the CI runs first.

**Comment by vshampor** [comment]
ONNX E2E 489 shows accuracy improvement for densenet12 (+0.77% acc.) and resnet50-v2-7 (+0.1% acc) and no difference for other models.

**Review by KodiaqQ — COMMENTED** [review]
General questions:

1. How does scale unification work in this PR? Based on the code, I assume that all possible layers unify now. Is that correct?
2. Why were most Tensorflow models updated in tests? What about other backends?
3. Were any conformance tests run on the final version? Are there any degradations?

**Comment by vshampor** [comment]
@KodiaqQ 

> General questions:
> 
> 1. How does scale unification work in this PR? Based on the code, I assume that all possible layers unify now. Is that correct?
> 2. Why were most Tensorflow models updated in tests? What about other backends?
> 3. Were any conformance tests run on the final version? Are there any degradations?

1. See the tests. The activation scales are unified based on the marker in HW config, and does not take into account metatypes.
2. I think your original PR with the changes in the same place also lead to TF model updates, so there should be no surprise here. 
3. See comment above ("PTQ build 240").

**Comment by KodiaqQ** [comment]
> 1. See the tests. The activation scales are unified based on the marker in HW config, and does not take into account metatypes.
> 2. I think your original PR with the changes in the same place also lead to TF model updates, so there should be no surprise here.
> 3. See comment above ("PTQ build 240").

Thanks! Let's run the validation again (with the latest changes in NNCF & OV) and verify the results. Then merge.

**Comment by vshampor** [comment]
@KodiaqQ PTQ build 285 shows a single regression vs build 286 on timm/visformer_small. Overall metric is still within 99% FP32.

**Review by KodiaqQ — APPROVED** [review]

**Comment by KodiaqQ** [comment]
> @KodiaqQ PTQ build 285 shows a single regression vs build 286 on timm/visformer_small. Overall metric is still within 99% FP32.

If the PTQ build is red, then we need to update the reference for the **timm/visformer_small** model as well. We can do it in this PR or follow-up, doesn't matter. But we should do it to keep builds green.
Also, is there any observation of why this model shows lower accuracy than the OV and ONNX versions?
